src/pulse_analyzer.c

```diff
@@ -273,40 +273,40 @@ void pulse_analyzer(pulse_data_t *data, int package_type)
         device.modulation  = OOK_PULSE_PPM; // TODO: there is not FSK_PULSE_PPM
         device.short_width = to_us * hist_gaps.bins[0].mean;
         device.long_width  = to_us * hist_gaps.bins[1].mean;
-        device.gap_limit   = to_us * (hist_gaps.bins[1].max + 1);                        // Set limit above next lower gap
-        device.reset_limit = to_us * (hist_gaps.bins[hist_gaps.bins_count - 1].max + 1); // Set limit above biggest gap
+        device.gap_limit   = to_us * (hist_gaps.bins[1].max + 1.0);                        // Set limit above next lower gap
+        device.reset_limit = to_us * (hist_gaps.bins[hist_gaps.bins_count - 1].max + 1.0); // Set limit above biggest gap
```

Comment:
Not great, it was int addition before, now it's double addition.

Reply:
Well, yes, the warning is about doing a 4 byte addition that then gets promoted to an 8 byte type for the multiplication, potentially leading to an overflow.
Using the floating point constant forces `double` computation which is not great, I agree. But how do I get rid of the warning then?

Reply:
This is non performance critical code. I'm fine with this change here but maybe it should be a cast? gap_limit is (float) while to_us is (double).